### PR TITLE
feat: Dashboard Config Option

### PR DESCRIPTION
**Pull request by ktx-vaidehi**
- #4589 

## Release Notes

- **New Features**
  - Enhanced dashboard panel configuration with new options for line thickness, label positioning, symbol visibility, and line interpolation.
  - Introduced new icons for visual representation of chart options, including Linear, NoSymbol, Smooth, StepBefore, StepAfter, and StepMiddle.

- **Bug Fixes**
  - Improved error handling and validation for dashboard panel configurations.

- **Documentation**
  - Added new localization entries for enhanced user interface labels related to dashboard features.

These updates significantly improve the customization and usability of dashboard panels, providing users with greater flexibility in visualizing their data.

### CODE REVIEW

**Review by ktx-kirtan — APPROVED** [review]

**Comment by ktx-kirtan** [comment]
@coderabbitai review

File: web/src/components/icons/dashboards/StepBefore.vue [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    viewBox="0 0 91 83"
+    fill="none"
+    xmlns="http://www.w3.org/2000/svg"
+  >
+    <path d="M6 76V5.5H47V76H86" stroke="currentColor" stroke-width="4" />
+    <circle cx="47" cy="6" r="6" fill="currentColor" />
+    <circle cx="85" cy="76" r="6" fill="currentColor" />
+    <circle cx="6" cy="77" r="6" fill="currentColor" />
+  </svg>
+</template>
```
[the remark this answers is not in the document]
Comment (ktx-kirtan) [reply web/src/components/icons/dashboards/StepBefore.vue:14]: This is not needed

File: web/src/components/icons/dashboards/Smooth.vue [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    <path
+      d="M6.07188 78C2.5731 -18.122 127.973 -20.5371 125.976 78"
+      stroke="currentColor"
+      stroke-width="4"
+    />
+    <circle cx="66" cy="6" r="6" fill="currentColor" />
+    <circle cx="126" cy="78" r="6" fill="currentColor" />
+    <circle cx="6" cy="78" r="6" fill="currentColor" />
+  </svg>
+</template>
```
[the remark this answers is not in the document]
Comment (ktx-kirtan) [reply web/src/components/icons/dashboards/Smooth.vue:18]: Not needed

File: web/src/components/icons/dashboards/StepMiddle.vue [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      fill="currentColor"
+    />
+    <circle
+      cx="6"
+      cy="6"
+      r="6"
+      transform="matrix(-1 0 0 1 26 71)"
+      fill="currentColor"
+    />
+  </svg>
```
[the remark this answers is not in the document]
Comment (ktx-kirtan) [reply web/src/components/icons/dashboards/StepMiddle.vue:24]: Not needed

File: web/src/components/icons/dashboards/StepAfter.vue [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  </svg>
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+
+export default defineComponent({
+  name: "StepAfter",
+});
+</script>
```
[the remark this answers is not in the document]
Comment (ktx-kirtan) [reply web/src/components/icons/dashboards/StepAfter.vue:40]: At all the places we are using this syntax, so need to use this only

File: web/src/components/icons/dashboards/StepAfter.vue [hunk 5]
```diff
@@ -0,0 +1,40 @@
+<template>
+  <svg
+    width="91"
+    height="83"
```
[the remark this answers is not in the document]
Comment (ktx-kirtan) [reply web/src/components/icons/dashboards/StepAfter.vue:4]: Not required

File: web/src/components/dashboards/addPanel/ConfigPanel.vue [hunk 6]
```diff
@@ -1246,6 +1465,70 @@ export default defineComponent({
...
+      },
+      {
+        label: t("dashboard.insideTop"),
+        value: "insideTop",
+      },
+      {
+        label: t("dashboard.insideBottom"),
+        value: "insideBottom",
+      },
+    ];
```
[the remark this answers is not in the document]
Comment (ktx-kirtan) [reply web/src/components/dashboards/addPanel/ConfigPanel.vue:1490]: Let this be here for now. We plan to refactor config module later

File: web/src/composables/useDashboardPanel.ts [hunk 7]
```diff
@@ -48,10 +48,17 @@ const getDefaultDashboardPanelData: any = () => ({
...
       top_results: null,
       top_results_others: false,
       axis_width: null,
       axis_border_show: false,
+      label_option: {
+        position: null,
+        rotate: 0,
+      },
+      show_symbol: false,
+      line_interpolation: "smooth",
```
[the remark this answers is not in the document]
Comment (ktx-kirtan) [reply web/src/composables/useDashboardPanel.ts:61]: Yes, we plan to do this with config refactor

File: src/common/meta/dashboards/v5/mod.rs [hunk 8]
```diff
@@ -204,6 +206,12 @@ pub struct PanelConfig {
     #[serde(skip_serializing_if = "Option::is_none")]
     axis_border_show: Option<bool>,
     #[serde(skip_serializing_if = "Option::is_none")]
+    label_option: Option<LabelOption>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    show_symbol: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    line_interpolation: Option<String>,
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply src/common/meta/dashboards/v5/mod.rs]: Updated with this changes

File: src/common/meta/dashboards/v5/mod.rs [hunk 9]
```diff
@@ -427,3 +435,11 @@
...
     pub unit: Option<String>,
 }
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, ToSchema)]
+pub struct LabelOption {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub position: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub rotate: Option<f64>,
+}
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply src/common/meta/dashboards/v5/mod.rs]: fixed with this

File: web/src/components/dashboards/addPanel/ConfigPanel.vue [hunk 10]
```diff
@@ -1020,6 +1207,14 @@
 import MarkLineConfig from "./MarkLineConfig.vue";
 import CommonAutoComplete from "@/components/dashboards/addPanel/CommonAutoComplete.vue";
 import CustomDateTimePicker from "@/components/CustomDateTimePicker.vue";
+import { getImageURL } from "../../../utils/zincutils";
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply web/src/components/dashboards/addPanel/ConfigPanel.vue]: removed

File: web/src/utils/dashboard/convertSQLData.ts [hunk 11]
```diff
@@ -1094,6 +1132,12 @@ export const convertSQLData = async (
             animation: false,
             data: getMarkLineData(panelSchema),
           },
+          label: {
+            show:
+              panelSchema.config?.label_option?.position == null ? false : true,
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply web/src/utils/dashboard/convertSQLData.ts]: changes are done

File: web/src/utils/dashboard/convertSQLData.ts [hunk 12]
```diff
@@ -1053,6 +1085,12 @@
           //     ?.color || "#5960b2",
           opacity: 0.8,
           ...defaultSeriesProps,
+          label: {
+            show:
+              panelSchema.config?.label_option?.position == null ? false : true,
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply web/src/utils/dashboard/convertSQLData.ts]: changes are done

File: web/src/utils/dashboard/convertSQLData.ts [hunk 13]
```diff
@@ -2304,11 +2371,20 @@
     case "area":
       return {
         type: "line",
-        smooth: true,
+        smooth:
+          panelSchema.config?.line_interpolation === "smooth" ||
+          panelSchema.config?.line_interpolation == null
+            ? true
+            : false,
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply web/src/utils/dashboard/convertSQLData.ts]: changes are done

File: web/src/utils/dashboard/convertSQLData.ts [hunk 14]
```diff
@@ -928,6 +944,14 @@
             name: panelSchema?.queries[0]?.fields?.y.find(
               (it: any) => it.alias == key
             )?.label,
+            label: {
+              show:
+                panelSchema.config?.label_option?.position == null
+                  ? false
+                  : true,
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply web/src/utils/dashboard/convertSQLData.ts]: changes are done

File: web/src/utils/dashboard/convertSQLData.ts [hunk 15]
```diff
@@ -897,11 +903,21 @@
...
               const seriesObj = {
                 //only append if yaxiskeys length is more than 1
                 name:
                   yAxisKeys.length == 1 ? key : key + " (" + yAxisName + ")",
                 ...defaultSeriesProps,
+                label: {
+                  show:
+                    panelSchema.config?.label_option?.position == null
+                      ? false
+                      : true,
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply web/src/utils/dashboard/convertSQLData.ts]: changes are done

File: web/src/utils/dashboard/convertSQLData.ts [hunk 16]
```diff
@@ -1296,6 +1340,12 @@
         const seriesObj = {
           name: key,
           ...defaultSeriesProps,
+          label: {
+            show:
+              panelSchema.config?.label_option?.position == null ? false : true,
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply web/src/utils/dashboard/convertSQLData.ts]: changes are done

File: web/src/utils/dashboard/convertSQLData.ts [hunk 17]
```diff
@@ -1021,6 +1045,14 @@
             name: panelSchema?.queries[0]?.fields?.y.find(
               (it: any) => it.alias == key
             )?.label,
+            label: {
+              show:
+                panelSchema.config?.label_option?.position == null
+                  ? false
+                  : true,
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply web/src/utils/dashboard/convertSQLData.ts]: changes are done

File: web/src/utils/dashboard/convertSQLData.ts [hunk 18]
```diff
@@ -2243,10 +2300,20 @@
       return {
         type: "line",
         emphasis: { focus: "series" },
-        smooth: true,
-        showSymbol: false,
+        smooth:
+          panelSchema.config?.line_interpolation === "smooth" ||
+          panelSchema.config?.line_interpolation == null
+            ? true
+            : false,
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply web/src/utils/dashboard/convertSQLData.ts]: changes are done

File: web/src/utils/dashboard/convertSQLData.ts [hunk 19]
```diff
@@ -2336,14 +2412,23 @@
     case "area-stacked":
       return {
         type: "line",
-        smooth: true,
+        smooth:
+          panelSchema.config?.line_interpolation === "smooth" ||
+          panelSchema.config?.line_interpolation == null
+            ? true
+            : false,
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply web/src/utils/dashboard/convertSQLData.ts]: changes are done

File: web/src/utils/dashboard/convertSQLData.ts [hunk 20]
```diff
@@ -722,6 +722,12 @@
         // inverse data for h-stacked and h-bar
         inverse: ["h-stacked", "h-bar"].includes(panelSchema.type),
         name: index == 0 ? panelSchema.queries[0]?.fields?.x[index]?.label : "",
+        label: {
+          show:
+            panelSchema.config?.label_option?.position == null ? false : true,
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply web/src/utils/dashboard/convertSQLData.ts]: changes are done

**Pull request merged**